esptfa_arima/Test_Management/views.py

```
@@ -336,20 +337,49 @@ class SubjectViewSet(viewsets.ModelViewSet):
 
 class QuarterViewSet(viewsets.ModelViewSet):
     queryset = Quarter.objects.all()
     serializer_class = QuarterSerializer
 
     # define the permissions
-    permission_classes = [permissions.IsAuthenticated]
+    permission_classes = [permissions.IsAuthenticated, IsTeacher]
+
+
+class TeacherAssignmentViewSet(viewsets.ModelViewSet):
+    queryset = TeacherAssignment.objects.all()
```

Ruff (0.14.10)

`TeacherAssignment` may be undefined, or defined from star imports

(F405)